src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -17,46 +21,58 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** 
-* Handles adding the form submission to the datastore in 
-* POST and returns json of the user's profile information
-* to client.
-*/
+/**
+ * Handles adding the form submission to the datastore in
+ * POST and returns json of the user's profile information
+ * to client.
+ */
 @WebServlet("/profile-data")
 public class ProfileServlet extends HttpServlet {
-  
   private static final String NAME_PROPERTY_KEY = "name";
   private static final String EMAIL_PROPERTY_KEY = "email";
   private static final String USERNAME_PROPERTY_KEY = "username";
   private static final String BIO_PROPERTY_KEY = "bio";
-  ProfileDatastore profileData = new ProfileDatastore();
+  private static final String EDIT_PROPERTY_KEY = "edit";
+  
+  private static final UserService userService = UserServiceFactory.getUserService();
+  private static final ProfileDatastore profileData = new ProfileDatastore();
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-
+    String email = userService.getCurrentUser().getEmail();
     String name = request.getParameter(NAME_PROPERTY_KEY);
-    String email = request.getParameter(EMAIL_PROPERTY_KEY);
-    String username = request.getParameter(USERNAME_PROPERTY_KEY );
+    String username = request.getParameter(USERNAME_PROPERTY_KEY);
     String bio = request.getParameter(BIO_PROPERTY_KEY);
+    boolean edit = Boolean.parseBoolean(request.getParameter(EDIT_PROPERTY_KEY));
```

Comment:
You should also check if the parameter given by ```EDIT_PROPERTY_KEY``` is valid and can be parsed into a boolean value.

Reply:
Boolean.parseBoolean() will return the value true if the string argument is not null and is equal, ignoring case, to the string "true". Otherwise it will return false. 

So we shouldn't need additional checks if we are okay with this logic.

Reply:
I don't think I need to do that because the value of edit is always true, which is a boolean